### PR TITLE
Add auto-links to player-data-purchasing.md

### PR DESCRIPTION
**Pull request by Katrist**
## Changes

Added auto-links to several items that were missing them.

## Checks

By submitting your pull request for review, you agree to the following:

- [X] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [X] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [X] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

File: content/en-us/scripting/data/player-data-purchasing.md [hunk 1]
```diff
@@ -143,7 +143,7 @@ Even though Roblox only allows a client to be connected to one server at a time,
 
 These scenarios should be rare, but they do occur, particularly in situations where a player disconnects from one server and connects to another in rapid succession (for example, while teleporting). Some malicious users might even attempt to abuse this behavior to complete actions without them persisting. This can be particularly impactful in games that allow players to trade and is a common source of item duplication exploits.
 
-Session locking addresses this vulnerability by ensuring that when a player's `Class.GlobalDataStore|DataStore` key is first read by the server, the server atomically writes a lock to the key's metadata inside the same `UpdateAsync` call. If this lock value is present when any other server attempts to read or write the key, the server does not proceed.
+Session locking addresses this vulnerability by ensuring that when a player's `Class.GlobalDataStore|DataStore` key is first read by the server, the server atomically writes a lock to the key's metadata inside the same `Class.GlobalDataStore:UpdateAsync()|UpdateAsync` call. If this lock value is present when any other server attempts to read or write the key, the server does not proceed.
```
Comment (IgnisRBX) [inline content/en-us/scripting/data/player-data-purchasing.md:146]:
```suggestion
Session locking addresses this vulnerability by ensuring that when a player's `Class.GlobalDataStore|DataStore` key is first read by the server, the server atomically writes a lock to the key's metadata inside the same `Class.GlobalDataStore:UpdateAsync()|UpdateAsync()` call. If this lock value is present when any other server attempts to read or write the key, the server does not proceed.
```

File: content/en-us/scripting/data/player-data-purchasing.md [hunk 2]
```diff
@@ -161,7 +161,7 @@ The transformation function passed into `Class.GlobalDataStore:UpdateAsync()|Upd
 
    - If this server has placed its own `LockId` value in the key's metadata previously, then this value is still in the key's metadata. This accounts for the situation where another server has taken over this server's lock (by expiry or by force) and later released it. Alternatively phrased, even if `LockId` is `nil`, another server could still have replaced and removed a lock in the time since you locked the key.
 
-2. `Class.GlobalDataStore:UpdateAsync()|UpdateAsync` performs the `Class.GlobalDataStore|DataStore` operation the consumer of `SessionLockedDataStoreWrapper` requested. For example, `GetAsync` translates to `function(value) return value end`.
+2. `Class.GlobalDataStore:UpdateAsync()|UpdateAsync` performs the `Class.GlobalDataStore|DataStore` operation the consumer of `SessionLockedDataStoreWrapper` requested. For example, `Class.GlobalDataStore:GetAsync()|GetAsync` translates to `function(value) return value end`.
```
Comment (IgnisRBX) [inline content/en-us/scripting/data/player-data-purchasing.md:164]:
```suggestion
2. `Class.GlobalDataStore:UpdateAsync()|UpdateAsync` performs the `Class.GlobalDataStore|DataStore` operation the consumer of `SessionLockedDataStoreWrapper` requested. For example, `Class.GlobalDataStore:GetAsync()|GetAsync()` translates to `function(value) return value end`.
```

File: content/en-us/scripting/data/player-data-purchasing.md [hunk 3]
```diff
@@ -175,7 +175,7 @@ A custom error message is also returned to the consumer, allowing the player dat
 
 ### Caveats
 
-The session locking regime relies on a server always releasing its lock on a key when it is done with it. This should always happen through an instruction to unlock the key as part of the final write in `Class.Players.PlayerRemoving|PlayerRemoving` or `OnGameClose`.
+The session locking regime relies on a server always releasing its lock on a key when it is done with it. This should always happen through an instruction to unlock the key as part of the final write in `Class.Players.PlayerRemoving|PlayerRemoving` or `Class.DataModel:BindToClose()|BindToClose`.
```
Comment (IgnisRBX) [inline content/en-us/scripting/data/player-data-purchasing.md:178]:
```suggestion
The session locking regime relies on a server always releasing its lock on a key when it is done with it. This should always happen through an instruction to unlock the key as part of the final write in `Class.Players.PlayerRemoving|PlayerRemoving` or `Class.DataModel:BindToClose()|BindToClose()`.
```

**Review by IgnisRBX — APPROVED** [review]

**Comment by IgnisRBX** [comment]
For the shorthand auto-links of methods (the text after the "|"), we prefer to keep the ending "()" after the method's name, to more clearly emphasize that it's a method, not a property or event. Thanks for the submission!

**Comment by Katrist** [comment]
I'll remember that next time, I just kept it the same for consistency. Thank you for the suggestion.

**Pull request merged**